lib/protocol/bep_extensions.go

```diff
@@ -126,6 +127,72 @@ func (f FileInfo) IsEmpty() bool {
 	return f.Version.Counters == nil
 }
 
+// IsEqual checks that the two file infos represent the same actual file.
+// Permissions (config) and blocks (scanning) can be excluded from the comparison.
+// Any file info is not "equal", if it has different
+//  - type
+//  - deleted flag
+//  - invalid flag
+//  - permissions, unless they are ignored
+// A file is not "equal", if it has different
+//  - modification time
+//  - size
+//  - blocks, unless there are no blocks to compare (scanning)
```

Comment:
Why don't we care about the version vector? I'm generally suspicious of this definition of "IsEqual". IMHO, "IsEqual" means that every attribute on the struct is exactly identical and nothing else. Possibly we can get away with `ignorePerms` and `ignoreBlocks`, but nothing else should be skipped unless we explicitly say so...

Reply:
I should have stuck with the original name `IsEquivalent` or maybe `Unchanged`. Version (and Sequence, potentially others) is purposely excluded. The job of this function is to identify file infos that represent the same underlying files (currently used for scanning and send-only resolution). The conditions have been take over from walk.go.